### ISSUE TITLE
Feature request: When multiple files into the timeline, they should be added one after the other, not overlapping

### ISSUE THREAD

**Issue by shodanx2** [OpenShot/openshot-qt#5190]
Here for example you can see

Two files placed one at a time, manually placing them one after the other with snapping

Then two files selected from Project Files and dropped together into the timeline, actually only one file is added, despite two being selected, it is always the first selected file

Then two the same two files, dropped from a windows explorer folder into the timeline, (this action also correctly added them to the project files when they were not present, however they were not added in alpha numerical order)

Lastly the same two files, added from explorer window, but with the other file selected for the drag and drop.

This action was repeated multiple times 

This is running openshot v2.6.1-release-candidate-8238-59878784-d588dbd6 build date 2021-09-04 23:49 (Please let me copy and paste this information from the about page)

![image](https://user-images.githubusercontent.com/10621885/230707854-0efb6a76-b541-4e03-be80-dc3e59d55d64.png)

**Comment by Colorjet3** [comment]
Hello @shodanx2.  I like this feature request and I will submit it to the dev team for review.  They will have the last say to approve it.

I wanted to follow up with one more question:
What should the behavior be if there are other clips on the timeline to the right of where you are dropping/importing the files?

There could be other scenarios as well.

Another option could be to somehow have some kind of an indicator on the top most clip that there are clips below it.

Can you think of other scenarios?

**Comment by Colorjet3** [comment]
Hello @shodanx2 Please submit additional ideas on how to handle things based on the scenario I presented. Things can get complex quickly when adding enhancements as there are more than one scenario to deal with.

I am going to close this ticket but you can continue to add to it.

**Comment by shodanx2** [comment]
Hello,

It seems I didn't reply to your earlier, I thought I did.

Usually if I were to drop multiple files into the timeline, 

I would like them to be added end-to-end on the timeline

They should be added at the exact location my cursor was when adding them.

If there are already clips in the way, they should be added "on-top of" but without adding transitions

After they are added they should all remain selected and ready to be moved immediately again

In the general case of two clips one top of one another, it should be visible that two clips are overlapping (Maybe 90% opacity so you can see through the top clip ?) (Maybe special coloration, coloration on the timeline "window" thing on top ?)

When clips are added you MIGHT OR MIGHT NOT want them to be attached to the cursor "as if" you were still holding down the mouse button after a click-and-drag.  This feature should be keyboard-toggeable. Cancellable with (ESC same as delete, the files don't get added, clicking actually deposits the files at the current location) 

I feel that's like a "sticky dragging" action, like dragging, but the drag is sticking to your cursor, as if you had a sticky mouse button maybe ? Anyway

When sticky dragging, the scroll mouse should have a special purpose, changing the order of the clips on the fly scrolling forward and the last clip become the first clip, first clip becomes second clip, second clip becomes third and so on, scrolling backwards, goes in the other direction. There should be a green line to show the beginning of the alphanumerically first file file and a red line to show the end of the alphanumerically last file. 

CTRL+scroll while stickydragging,  
CTRL+Scroll forward while sticky dragging order clips in filename alphanumerical ascending order 
CTRL+Scroll backward while sticky dragging order clips in filename alphanumerical descending order
CTRL+Scroll forward while sticky dragging order clips in file creation date ascending order 
CTRL+Scroll backward while sticky dragging order clips in file creation date descending order 
CTRL+Scroll forward twice in a row while sticky dragging order clips in file modification date ascending order 
CTRL+Scroll backward twice in a row while sticky dragging order clips in file modification date descending order 
CTRL+Alt+Scroll forward while sticky dragging, order clips from shortest to longest (either actual clip length, or if that's too 
CTRL+Shift+Scroll forward while sticky dragging, order clips from shortest to longest (either actual clip length, or if that's too hard, by filesize)
CTRL+Shift+Scroll forward while sticky dragging, order clips from longest to shortest 


That reminds me, can we apply arbitrary colours to clips, or color them by group or category ? (I would love to see ma CAM1 CAM2 CAM3 groups however I have not investigated if this is possible)

As far as the organizational elements, I never use the "Project Files" I prefer dropping files from explorer, I have an explorer windows with the Always On Top flag set.


I would like to be able to drop more things in there

Entire folders
Links
LLC files (WOULD BE FUCKING AWESOME) (these are the files produced by Losslesscut, an amazing piece of software that I use to scrub, rough cut and merge my video files before I go into openshot, I sometime do the entire job in losslesscut, I will wish there were a losslesscrop even if it's not really lossless)   Anyway, the LLC files describe Multiple cut points from a single file, it is  ma-gi-cal
Also, I don't know if Openshot does it, but when openshot project file is saved, are the paths absolute or relative ?

Imagine the following folder structure, imagine I want to drag 

```
c:\my video project\openshot\rough1.osp
c:\my video project\openshot\scene1.osp
c:\my video project\openshot\scene2.osp
c:\my video project\openshot\scene2-angle2.osp
c:\my video project\openshot\scene2-angle2v2.osp
c:\my video project\openshot\scene2-angle2v3.osp
c:\my video project\openshot\final1-1.osp
c:\my video project\openshot\final1-v1.1.osp
c:\my video project\openshot\final2.osp
c:\my video project\rough1.osp.lnk
c:\my video project\scene1.osp.lnk
c:\my video project\scene2.osp.lnk -> c:\my video project\openshot\scene2-angle2v3.osp
c:\my video project\final.osp.lnk -> c:\my video project\openshot\final2.osp
c:\my video project\final.mp4.lnk -> c:\my video project\final1\final.mp4
c:\my video project\cam1.llc -> c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-cut-merged-1666487677781 complet moins face-proj.llc
c:\my video project\original\cam1\vid 2023-04-22 22h44s01.mp4
c:\my video project\original\cam1\vid 2023-04-22 22h54s02.mp4
c:\my video project\original\cam1\vid 2023-04-22 23h04s03.mp4
c:\my video project\original\cam2\vid 2023-04-22 22h44s10.mp4
c:\my video project\original\cam2\vid 2023-04-22 22h54s11.mp4
c:\my video project\original\cam10\360 2023-04-22 22h44s15.mp4
c:\my video project\original\cam10\360 2023-04-22 22h54s15.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-cut-merged-1666487677781 complet moins face-proj.llc
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-cut-merged-1666487677781 complet moins face-proj.llc
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-cut-merged-1666487677781 complet moins face-00.00.00.000-00.06.28.778-merged.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-cut-merged-1666487677781 complet moins face-00.00.00.000-00.06.28.778.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-cut-merged-1666487677781 complet moins face.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-00.06.26.743-00.06.29.910-seg13.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-00.04.01.371-00.06.29.910-seg12.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-00.03.55.845-00.04.01.371-seg11.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-00.03.07.208-00.03.53.333.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-00.03.07.089-00.03.55.845-seg10.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 211907850-00.03.01.952-00.03.06.935-seg9.mp4
c:\my video project\precut\cam1\vid 2023-04-22 22h44s01 11907850-proj
c:\my video project\scene1\cam1\vid 2023-04-22 22h44s01 211907850-00.04.01.371-00.06.29.910-seg12.lnk
c:\my video project\scene1\cam1\vid 2023-04-22 22h44s01 211907850-00.03.55.845-00.04.01.371-seg11.lnk
c:\my video project\scene1\cam1\vid 2023-04-22 22h44s01 211907850-00.03.07.208-00.03.53.333.lnk
c:\my video project\scene1\cam1\vid 2023-04-22 22h44s01 211907850-00.03.07.089-00.03.55.845-seg10.lnk
c:\my video project\final1\final.mp4
```

**Comment by Colorjet3** [comment]
Hello @shodanx2 Great feedback and thank you!  I am to link this response to the internal enhancement card.

Have a great day!
